src/boot/efi/boot.c

```diff
@@ -104,6 +104,11 @@ enum {
         TIMEOUT_TYPE_MAX    = UINT32_MAX,
 };
 
+enum {
+        IDX_MAX = INT16_MAX,
+        IDX_INVALID,
```

Comment:
we usually initialize the _INVALID enum to `-EINVAL`

Reply:
Yes, please do this so that it can _INVALID can be used in strerror().

Reply:
Hmm, so here IDX_INVALID is positive, and the return type is changed to UINTN. Why do this change? It'd seem to make things more confusing.

Reply:
This isn't userspace, so no `strerror()`. And UINTN (== size_t) is used everywhere for array indices and lengths. Using a signed type is more confusing and inconsistent. It also creates unsound signed/unsigned comparisons that require a cast.

If you want to insist, I can set this to UINTN_MAX (which was the effective invalid value before). Ultimately, the value doesn't matter as it's just an internal sentinel.

Reply:
OK, the enum here is just to give a name to the constant.